### PR TITLE
README.md: add 'About the Lesson' section

### PR DESCRIPTION
**Pull request by Josh-Woodcock**
#495 Improve README.md

Based [swcarpentry/r-novice-inflammation]([https://github.com/swcarpentry/r-novice-inflammation/blob/gh-pages/README.md) README

- put maintainers in a list, easier to read
- add a short introduction
- add links to other language versions of lesson (R and MATLAB)

### CODE REVIEW

File: README.md [hunk 1]
```diff
@@ -1,10 +1,32 @@
 # Programming with Python
+
```
Comment (maxim-belkin) [inline README.md]: 👍

File: README.md [hunk 2]
```diff
@@ -1,10 +1,32 @@
...
 
+Current maintainers:
+
+* [Trevor Bekolay][trevor_bekolay]
+* [Maxim Belkin][maxim_belkin]
+* [Anne Fouilloux][anne_fouilloux]
+* [Valentina Staneva][valentina_staneva]
+* [Mike Trizna][mike_trizna]
+* [Greg Wilson][greg_wilson], creator of [Software Carpentry][swc_history]
+
```
Comment (maxim-belkin) [inline README.md]: 👎 on these (with the exception of the last change - `Greg Wilson, creator of ...`). This section was worded carefully and list takes a lot of vertical space.

File: README.md [hunk 3]
```diff
@@ -1,10 +1,32 @@
...
+The goal of this lesson is to teach novice programmers to write modular code to
```
Comment (maxim-belkin) [inline README.md]: Let's not say what the goal of the lesson is. Let's just say that this lesson teaches <blah>.

File: README.md [hunk 4]
```diff
@@ -1,10 +1,32 @@
...
+perform a data analysis. Python is used to teach these skills because it is a
+commonly used programming language in many scientific disciplines. However, the
```
Comment (maxim-belkin) [inline README.md]: It is not the only reason why we use Python, so I suggests leaving this statement out.

File: README.md [hunk 5]
```diff
@@ -1,10 +1,32 @@
...
+emphasis is not on teaching every aspect of Python, but instead on
```
Comment (maxim-belkin) [inline README.md]: Let's not make negative statements about the emphasis :)

File: README.md [hunk 6]
```diff
@@ -1,10 +1,32 @@
...
+language agnostic principles like automation with loops and encapsulation with
```
Comment (maxim-belkin) [inline README.md]: language-agnostic

File: README.md [hunk 7]
```diff
@@ -1,10 +1,32 @@
...
+functions (see [Best Practices for Scientific Computing][best-practices] to
+learn more). This lesson is also available in [R][R]
```
Comment (maxim-belkin) [inline README.md]: `This lesson...` deserves a separate paragraph

File: README.md [hunk 8]
```diff
@@ -1,10 +1,32 @@
...
+and [MATLAB][MATLAB].
+
+The example used in this lesson analyzes a set of 12 data files with
+inflammation data collected from a trial for a new treatment for arthritis (the data was simulated). Learners are shown how it is better to create a function and apply it to each of the 12 files using a loop instead of using copy-paste to analyze the 12 files individually.
```
Comment (maxim-belkin) [inline README.md]: I like the idea of having such a paragraph, but not the current wording of it.

File: README.md [hunk 9]
```diff
@@ -32,3 +50,6 @@ To cite this lesson, please consult with [CITATION](CITATION)
 [valentina_staneva]: http://software-carpentry.org/team/#staneva_valentina
 [greg_wilson]: https://github.com/gvwilson
 [swc_history]: https://software-carpentry.org/scf/history/
+[best-practices]: http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745
+[R]: https://github.com/swcarpentry/r-novice-inflammation
+[MATLAB]: https://github.com/swcarpentry/matlab-novice-inflammation
```
Comment (maxim-belkin) [inline README.md]: 👍 would be good to reference also "good enough" practices :)

File: README.md (hunk 8 above)
Comment (maxim-belkin) [reply README.md]: Also, it might be useful to add a small title like `About the lesson` or similar

Comment (annefou) [reply README.md]: I like the idea @Josh-Woodcock had to give an overview of what the lesson is about (like for the R lesson in the README file). 

And @maxim-belkin your suggestion to add "About the lesson" will help to highlight this paragraph.

File: README.md (hunk 9 above)
Comment (maxim-belkin) [reply README.md]: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510

File: README.md [hunk 10]
```diff
@@ -39,3 +51,14 @@ To cite this lesson, please consult with [CITATION](CITATION)
...
 [swc_history]: https://software-carpentry.org/scf/history/
+<<<<<<< HEAD
+<<<<<<< HEAD
+[best-practices]: http://journals.plos.org/plosbiology/article?id=10.1371/journal.pbio.1001745
+=======
+[good-enough-practices]: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510
+>>>>>>> - added "About the Lesson" section.
+=======
+[good-enough-practices]: http://journals.plos.org/ploscompbiol/article?id=10.1371/journal.pcbi.1005510
+>>>>>>> - Cleaned up formatting
```
Comment (maxim-belkin) [inline README.md]: Please resolve the conflicts.

File: README.md [hunk 11]
```diff
@@ -1,12 +1,24 @@
...
 [![GitHub release](https://img.shields.io/github/release/swcarpentry/python-novice-inflammation.svg)](https://github.com/swcarpentry/python-novice-inflammation/releases)
 [![Create a Slack Account with us](https://img.shields.io/badge/Create_Slack_Account-The_Carpentries-071159.svg)](https://swc-slack-invite.herokuapp.com/)
 [![Slack Status](https://img.shields.io/badge/Slack_Channel-swc--py--inflammation-E01563.svg)](https://swcarpentry.slack.com/messages/C9Y0L6MF0)
 
+
+## About the Lesson
+
 An introduction to Python for non-programmers using inflammation data.
 
+This lesson teaches novice programmers to write modular code to perform data analysis using Python. However, the emphasis is on teaching language-agnostic principles such as automation with loops and encapsulation with functions (see [Good enough practices in scientific computing][good-enough-practices] to
```
Comment (maxim-belkin) [inline README.md]: Suggestion:

> The emphasis, however is on teaching fundamental and, therefore, language-agnostic principles of programming such as...

File: README.md [hunk 12]
```diff
@@ -1,12 +1,24 @@
...
+learn more).
+
+The example used in this lesson analyses a set of 12 data files with inflammation data collected from a trial for a new treatment for arthritis (the data was simulated). Learners are shown how it is better to create a function and apply it to each of the 12 files using a loop instead of using copy-paste to analyse the 12 files individually.
```
Comment (maxim-belkin) [inline README.md]: `the data was simulated` ~> `simulated data`

Comment (maxim-belkin) [reply README.md]: I think the last sentence could be improved. Don't mention `a loop`. Use phrases like `Python enables ...` and `automation` when talking about Python, and , `error-prone` (or similar) when talking about analysis done "by hand".

**Comment by maxim-belkin** [comment]
Thank you very much, @Josh-Woodcock!

**Pull request merged**